agent/agent-tooling/src/main/java/com/microsoft/applicationinsights/agent/internal/init/SecondEntryPoint.java

```diff
@@ -264,13 +262,27 @@ public void customize(AutoConfigurationCustomizer autoConfiguration) {
         .addPropertiesCustomizer(new AiConfigCustomizer())
         .addLogRecordProcessorCustomizer(
             (logRecordProcessor, configProperties) -> {
-              if (firstLogRecordProcessor.getAndSet(false)) {
-                // hack to run our log record processors first, before any other log processors
-                // (in particular before the batch log processor which performs the export)
-                // see https://github.com/open-telemetry/opentelemetry-java/issues/6599
+              if (logRecordProcessor instanceof BatchLogRecordProcessor) {
```

Comment:
Could you please add a comment about why things have to be done at the `BatchLogRecordProcessor` level.